.github/workflows/release.yaml

```diff
@@ -37,3 +39,32 @@ jobs:
         uses: helm/chart-releaser-action@v1.4.1
         env:
           CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+      # see https://github.com/helm/chart-releaser/issues/183
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: sigstore/cosign-installer@v2.8.1
+      - name: Push charts to GHCR
+        env:
+          COSIGN_EXPERIMENTAL: 1
+        # when filling gaps with previously released charts, cr would create
+        # nothing in .cr-release-packages/, and the original globbing character
+        # would be preserved, causing a non-zero exit. Set nullglob to fix this
+        run: |
+          shopt -s nullglob
+          for pkg in .cr-release-packages/*; do
+            if [ -z "${pkg:-}" ]; then
+              break
+            fi
+            helm push "${pkg}" oci://ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts
+            file=${pkg##*/}
+            name=${file%-*}
+            version=${file%.*}
+            version=${version#*-}
+            cosign sign ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts/"${name}":"${version}"
```

Comment:
Hi @scottrigby, instead of signing the tags, it'd be better to use digests to provide a better assurance about the content we are signing. I did a similar task in the Kyverno project, as you can see from [here](https://github.com/kyverno/kyverno/blob/5557d6c85d7e1866ebd3bed5c1c3a4c9d55a3881/.github/workflows/helm-release.yaml#L93). 

**TLDR**;

_AFAIK, the `helm push` command outputs the digest of the image being pushed, so we can save that digest in a file and retrieve it from there later._

WDYT?

Reply:
Kindly ping @monotek @torstenwalter, do you have any comments about it?

Reply:
_https://github.com/prometheus-community/helm-charts/pull/2664_